corehq/apps/reports/views.py

```diff
@@ -1771,6 +1771,14 @@ def _error(msg):
             non_parents = filter(lambda cb: cb.path == [], case_blocks)
             if len(non_parents) == 1:
                 edit_session_data['case_id'] = non_parents[0].caseblock.get(const.CASE_ATTR_ID)
+                case = CaseAccessors(domain).get_case(edit_session_data['case_id'])
+                if case.closed:
+                    return _error(_(
+                        u'Case {} is closed. Please reopen the '
+                        u'case before editing the form'
+                    ).format(case.case_id))
+                elif case.deleted:
+                    return _error(_(u'Case {} is deleted. Cannot edit this form.').format(case.case_id))
```

Comment:
i think we show a 404 when we link to a deleted case, but i cleaned up the UX for archived case https://github.com/dimagi/commcare-hq/pull/15133/commits/3f3ffacd710a58159b1e460625440e24ffc6a8d9

<img width="1659" alt="screen shot 2017-02-27 at 11 45 29 am" src="https://cloud.githubusercontent.com/assets/918514/23356454/84ec680e-fce2-11e6-847b-9fd601eb2d47.png">